File: src/main/zig/2020/runner.zig
const aoc = @import("../aoc.zig");

pub fn run(problem: *aoc.Problem, day: u16) !aoc.Solution {
    return try switch (day) {
        1 => @import("day01.zig").run(problem),
        2 => @import("day02.zig").run(problem),
        3 => @import("day03.zig").run(problem),
        4 => @import("day04.zig").run(problem),
        5 => @import("day05.zig").run(problem),
        6 => @import("day06.zig").run(problem),
        7 => @import("day07.zig").run(problem),
        8 => @import("day08.zig").run(problem),
        9 => @import("day09.zig").run(problem),
        10 => @import("day10.zig").run(problem),
        11 => @import("day11.zig").run(problem),
        12 => @import("day12.zig").run(problem),
        13 => @import("day13.zig").run(problem),
        14 => @import("day14.zig").run(problem),
        15 => @import("day15.zig").run(problem),
        16 => @import("day16.zig").run(problem),
        17 => @import("day17.zig").run(problem),
        18 => @import("day18.zig").run(problem),
        19 => @import("day19.zig").run(problem),
        20 => @import("day20.zig").run(problem),
        22 => @import("day22.zig").run(problem),
        else => unreachable,
    };
}

File: src/main/nim/aoc/2017/runner.nim
import ../aoc
import day01, day02, day03, day04, day05, day06, day07, day08, day09, day10, day11, day12, day13, day14, day15, day16, day17, day18, day19, day20, day21, day22, day24, day25

proc run2017*(file: string, day: int): Solution =
  case day
  of 1: day01(file)
  of 2: day02(file)
  of 3: day03(file)
  of 4: day04(file)
  of 5: day05(file)
  of 6: day06(file)
  of 7: day07(file)
  of 8: day08(file)
  of 9: day09(file)
  of 10: day10(file)
  of 11: day11(file)
  of 12: day12(file)
  of 13: day13(file)
  of 14: day14(file)
  of 15: day15(file)
  of 16: day16(file)
  of 17: day17(file)
  of 18: day18(file)
  of 19: day19(file)
  of 20: day20(file)
  of 21: day21(file)
  of 22: day22(file)
  of 24: day24(file)
  of 25: day25(file)
  else: raise newException(ValueError, "Invalid day")
